scripts/asr_language_modeling/ngram_lm/eval_beamsearch_ngram_transducer.py

```diff
@@ -0,0 +1,273 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This script would evaluate an N-gram language model trained with KenLM library (https://github.com/kpu/kenlm) in
+# fusion with beam search decoders on top of a trained ASR model. NeMo's beam search decoders are capable of using the
+# KenLM's N-gram models to find the best candidates. This script supports both character level and BPE level
+# encodings and models which is detected automatically from the type of the model.
+# You may train the LM model with 'scripts/ngram_lm/train_kenlm.py'.
+#
+# USAGE: python eval_beamsearch_ngram.py  --nemo_model_file=<path to the .nemo file of the model> \
+#                                         --input_manifest=<path to the evaluation JSON manifest file \
+#                                         --kenlm_model_file=<path to the binary KenLM model> \
+#                                         --beam_width=<list of the beam widths> \
+#                                         --beam_alpha=<list of the beam alphas> \
+#                                         --preds_output_folder=<optional folder to store the predictions> \
+#                                         --decoding_mode=maes
+#                                         ...
+#
+# You may find more info on how to use this script at:
+# https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/main/asr/asr_language_modeling.html
+
+
+import argparse
```

Comment:
Instead of two different script for beam search, would it be possible to use high level beam search API of RNNT and merge this script whth the one we already have?

Reply:
I will try to check this possibility

Reply:
@titu1994, I tried to merge RNNT beam search decoding into `eval_beamsearch_ngram_transducer.py`, but it looks a bit overloaded due to different decoding parameters and methods for CTC and RNNT. Furthermore, we need to add the option of logits returning for rnnt_model.transcribe method. We can stop at the current version of the two different files for this PR. I can try to merge them in the future separate PR.

Reply:
Transducer does not compute full logits, you can get back alignment dangling matrix with preserve alignments and return hypothesis so nj need for additional logprobs flag. 

I suppose we can keep separate script for now though. Let me know when PR is finalized

Reply:
I meant RNNT encoder logits which are used for `asr_model.decoding.rnnt_decoder_predictions_tensor` by the analogy with `model.decoding.ctc_decoder_predictions_tensor` as in `eval_beamsearch_ngram_transducer.py` script.

Reply:
I get that, but adding an argument to RNNT transcribe just for that chances signature of RNNT and CTC. You can use logprobs flag but you won't get actual logprobs back but encoder output. Please use hypothesis output for encoder logits or repurpose logprobs as encoder output

Reply:
Ok, I will try to do it in another PR. 
Could you check the current PR and approve it if everything is OK? I finalized all the work.